DataFormats/Math/interface/alpaka/deltaPhi.h

```diff
@@ -0,0 +1,35 @@
+#ifndef DataFormats_Math_alpaka_deltaPhi_h
+#define DataFormats_Math_alpaka_deltaPhi_h
+
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+
+namespace cms::alpakatools {
+
+  // reduce to [-pi,pi]
+  template <typename TAcc, typename T>
+  ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE constexpr T reduceRange(TAcc const& acc, T x) {
+    constexpr T o2pi = 1. / (2. * M_PI);
```

Comment:
Now that we have C++20, how about
```suggestion
    constexpr T o2pi = T{1.} / (T{2.} * std::numbers::pi_v<T>);
```
?

Is it intentional to do the calculations on double and cast the result to `T`?

Is `T` expected to be only a floating point type?

Reply:
I think that it would be better to maintain the implementation as close as possible to the original `deltaPhi.h`

Reply:
> That said, your suggestion makes sense but maybe it should be propagated to `interface/deltaPhi.h` as well? Not sure whether we want to do so though (it was not the initial purpose of this PR).

Ok, maybe it would make sense to keep both versions consistent, and maybe any changes would be beyond this PR.

> I suppose that the intention is that `T` is only a floating point type but I am not sure. For LST purposes (for which these functions are going to be used, at least initially), floating point type only `T` is enough.

I was wondering if anyone would be interested to add that as a type constraint either as
```cpp
  template <typename TAcc, typename T>
  requires std::is_floating_point_v<T>
  ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE T reduceRange(TAcc const& acc, T x) {
```
or
```cpp
  template <typename TAcc>
  ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE std::floating_point auto reduceRange(TAcc const& acc, std::floating_point auto x) {
```
but probably that's beyond this PR as well.

Reply:
Note that doing the computation in double precision may be quite expensive on most GPUs. For example it is 64 times more expensive than single precision on an NVIDIA L4.

Reply:
I will go ahead and apply the suggestion by Matti to fix this:
```constexpr T o2pi = T{1.} / (T{2.} * std::numbers::pi_v<T>);```